### ISSUE TITLE
Provide ngDebugContext in production mode

### ISSUE THREAD

**Issue by bengourley** [angular/angular#27752]
# 🚀 feature request

### Relevant Package
This feature request is for `@angular/core`

### Description
I work on the Angular integration for Bugsnag. In development we have access to the really useful `ngDebugContext` object which we make use of to include Angular-specific metadata to error reports: https://github.com/bugsnag/bugsnag-js/blob/next/packages/plugin-angular/src/index.ts#L27-L30.

This object is never present when an app is run in production mode.

### Describe the solution you'd like
Would it be possible to provide this object in production?

### Describe alternatives you've considered
I suspect it is not provided because it is expensive to do so. Is there something less expensive, but still potentially useful for debugging error reports that can be supplied? Or is there anything else that already exists that I haven't discovered which can provide info on what component/service etc. may have errored?

Thanks!

**Comment by pkozlowski-opensource** [comment]
With the ivy rewrite the `ngDebugContext` will most likely go away but we are going to have a replacement that will be available in both dev and prod modes. We are not going to modify the existing view engine but you should have what you need in ivy. Stay tuned.

**Comment by jacano** [comment]
News on this feature?

**Comment by pkozlowski-opensource** [comment]
Update for my [last comment](https://github.com/angular/angular/issues/27752#issuecomment-448628143): we've got more info in the debug mode with ivy, but those are still development mode only (and we've got no intention of making those available in prod as it would slow down your application).

The way I understand the problem is that you would like to, in case of an error / exception, to "easily" get access to:
* a component / template that miss-behaved;
* a context (data) against which the mentioned template was evaluated.

With ivy stack traces are the whole lot more useful (as they directly point to a template in question) but a stack trace won't give you evaluation context.

I guess we could unify exceptions thrown internally (and caught by the `ErrorHandler`) to include more info so identifying a component / template in question is easier.
